src/microbe_stage/MicrobeAI.cs

```diff
@@ -14,6 +15,7 @@
 /// </remarks>
 public class MicrobeAI
 {
+    // ReSharper disable once NotAccessedField.Local
     private readonly Compound atp;
```

Comment:
If this is not accessed, this field should be deleted. Though I'm curious as it seems that this being unused means that the AI no longer knows to stop moving if it is out of ATP to keep from dying?

Reply:
Fields removed; I was following the pattern set in this class by the iron variable. You may want to creating constants for these pointers and migrating to using them, since these are used in many places.

The RunAndTumble algorithm includes a chance of pausing when it thinks the microbe is in nutrients, but it was an intentional choice to not add a "give up and hope glucose falls into you" clause to permanently stop moving. When observing microbes in playtesting, the case of not being able to sustain movement but still having enough compounds to keep homeostasis was quite rare. What I've seen is that when a microbe is starving, it's either because it made bad choices and didn't look for food, or it needs to move forward and get to the next cloud.

Reply:
They are loaded from JSON, the reason this approach is used instead of just using the internal names like "atp" is that this way the code explodes if some compound is renamed, which is unlikely but possible.
Also consider if there exists a mod that adds new compounds and that can be loaded / unloaded, it just doesn't work having constants for the compounds or the other stuff that is loaded from JSON.

Reply:
> The RunAndTumble algorithm includes a chance of pausing when it thinks the microbe is in nutrients, but it was an intentional choice to not add a "give up and hope glucose falls into you" clause to permanently stop moving. When observing microbes in playtesting, the case of not being able to sustain movement but still having enough compounds to keep homeostasis was quite rare. What I've seen is that when a microbe is starving, it's either because it made bad choices and didn't look for food, or it needs to move forward and get to the next cloud.

That's a vital strategy for plant cells, as they need to have a ton of chloroplasts to move constantly. I'm sure @Untrustedlife is not happy that plant cells are neglected by this new AI.

Reply:
I've been acting under the assumption that all microbes are trying to reproduce (i.e. there are no "prop" microbes that only exist for the player to kill). In the current game, there is no way to do that without phosphorus and ammonia, and no way to get those without moving constantly, save for hoping lots of aggressive, dumb cells get themselves killed right next to your position.

I know it breaks our concept of a plant, but I'm working on observations here. I've played and seen others play photosynthesizers that move, and they're quite viable. I have not seen a stationary plant reproduce, in the old or new AI. I can put in sessile behavior very easily, but I don't think I can make one that doesn't instantly go extinct or rely exclusively on auto-evo to keep existing.

Reply:
The plan is to add currents and other features that make immobile plant cells viable, the AI should be made now so that it makes sense for plants as well. The AI can make the assumption that plant cells are successful off-screen.

Reply:
Easy enough. Should plants still try to run from predators? What about if they are currently being engulfed?

Reply:
I think most cells should try to run when someone tries to attack them (if they have no way to defend themselves) otherwise predator gameplay is way too strong. I think @TjWhale may have done an AI version that *just* ran away from the player, and even that made being a predator more interesting as cells wouldn't just suicide into you.

Reply:
I've now added a conditional on the run and tumble behavior: https://github.com/adQuid/Thrive/blob/improve_microbe_ai/src/microbe_stage/MicrobeAI.cs#L99

This allows low-activity microbes to still flee, eat chunks, or even hunt if the other personality traits happen to be high enough.